docs/ref/databases.txt

```diff
@@ -216,6 +216,48 @@ configuration in :setting:`DATABASES`::
 
 .. _isolation level: https://www.postgresql.org/docs/current/transaction-iso.html
 
+Role setting
+------------
+The PostgreSQL permission model consists of roles, which can represent
+users or groups. Each database object (table, sequence, view, etc...)
+is assigned an owning role and that owner is the only role that can
+modify or drop those objects. If the Django application is running using
+temporary credentials from an external credential management system then
+that temporary role will not match the owning role for the database
+objects and migrations or other structural mutations to the database
+objects will fail with permission errors.
+
+To solve this, create a role that acts as a group within the database
+that can own those objects then create your individual user roles as
+members of that owning role.::
+
+  CREATE ROLE my_application_role;
+  CREATE ROLE temporary_user_role IN ROLE my_application_role;
+
+Setting the ``assume_role`` option in the :setting:`OPTIONS` of your
+database configuration in :setting:`DATABASES`. This will cause Django
+to issue a ``SET ROLE`` statement during connection creation which will
+cause the logged-in user to assume the group role which has owner access
+to those database objects.
+
+.. code-block:: python
+    :caption: ``settings.py``
+
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql',
+            'OPTIONS': {
+                'assume_role': 'my_application_role',
+            },
+        }
+    }
+
+.. note::
+
+   If the database was created before the introduction of a grouping role then
+   object ownership may be inconsistent which will result in permission errors
+   until the ownership of each object is updated.
```

Comment:
IMO, Django docs are not the right proper place to describe solutions to OPS issues. I'd limit this section to a brief description of what and how can be set, e.g.

```
If you need to use a role for database connections, set it in the
:setting:`OPTIONS` part of your database configuration in
:setting:`DATABASES`::

    DATABASES = {
        # ...
        "OPTIONS": {
            "assume_role": "my_application_role",
        },
    }
```

Reply:
Wasn't quite sure what the right balance was here. I've revised the docs to be closer in-line with your proposal. WDYT?